### PR TITLE
check if the project has the userMetadata

### PR DESCRIPTION
**Pull request by jackyq2015**
@thadguidry This is to fix the bug you found today that the index page will be blank after creating the userMetadata from the preference page.

### CODE REVIEW

**Comment by jackyq2015** [comment]
Will put a filter when user define the userMetadata

**Comment by wetneb** [comment]
@thadguidry again I think you should wait for a programmer to review the code before merging! Your reviews are very useful for the user standpoint, but we also need to watch for code quality. We really need to stick to this rules, otherwise we are just going to ruin the codebase. I am not saying I should have the last word on every PR: any coder with some familiarity with the code can do it (e.g. @ostephens).

**Comment by thadguidry** [comment]
@wetneb Jacky and I would like to talk with you offline in a chat sometime.  Do you have time this weekend ?

**Comment by wetneb** [comment]
@thadguidry sure! preferably in your morning so that it's not in the middle of my night :)

File: main/webapp/modules/core/scripts/preferences.js [hunk 1]
```diff
@@ -50,6 +50,13 @@ dictionary = data;
 $.i18n.setDictionary(dictionary);
 //End internationalization
 
+function deDupUserMetaData(arrObj)  {
+    var result = _.uniq(JSON.parse(arrObj), function(x){
+        return x.name;
+    });
+    
+    return JSON.stringify(result).replace(/"/g, '\"');
```
Comment (wetneb) [inline main/webapp/modules/core/scripts/preferences.js]: Why this `.replace(/"/g, '\"')`?

**Pull request merged**